reporter/sbom/spdx.go

```diff
@@ -0,0 +1,587 @@
+			packages = append(packages, &libpkgs[0])
+			relationships = append(relationships, makeSPDXRelationShip(root.PackageSPDXIdentifier, libpkgs[0].PackageSPDXIdentifier, relationShipContains))
+			for _, pack := range libpkgs[1:] {
+				packages = append(packages, &pack)
+				relationships = append(relationships, makeSPDXRelationShip(libpkgs[0].PackageSPDXIdentifier, pack.PackageSPDXIdentifier, relationShipDepensOn))
+			}
+		}
+	}
+
+	for _, ghm := range result.GitHubManifests {
+		if ghpkgs := ghpkgToSPDXPackages(ghm, packageToURLMap, result.ReportedAt); len(ghpkgs) > 0 {
+			packages = append(packages, &ghpkgs[0])
+			relationships = append(relationships, makeSPDXRelationShip(root.PackageSPDXIdentifier, ghpkgs[0].PackageSPDXIdentifier, relationShipContains))
+			for _, pack := range ghpkgs[1:] {
+				packages = append(packages, &pack)
+				relationships = append(relationships, makeSPDXRelationShip(ghpkgs[0].PackageSPDXIdentifier, pack.PackageSPDXIdentifier, relationShipDepensOn))
+			}
+		}
+	}
+
+	if wppkgs := wppkgToSPDXPackages(result.WordPressPackages, packageToURLMap, result.ReportedAt); len(wppkgs) > 0 {
+		packages = append(packages, &wppkgs[0])
+		relationships = append(relationships, makeSPDXRelationShip(root.PackageSPDXIdentifier, wppkgs[0].PackageSPDXIdentifier, relationShipContains))
+		for _, pack := range wppkgs[1:] {
+			packages = append(packages, &pack)
+			relationships = append(relationships, makeSPDXRelationShip(wppkgs[0].PackageSPDXIdentifier, pack.PackageSPDXIdentifier, relationShipDepensOn))
+		}
+	}
+
+	return packages, relationships
+}
+
+func ospkgToSPDXPackages(r models.ScanResult, packageToURLMap map[string][]string) []spdx.Package {
+	if r.Family == "" || len(r.Packages) == 0 {
+		return nil
+	}
+
+	type srcpkg struct {
+		name    string
+		version string
+		arch    string
+	}
+	binToSrc := map[string]srcpkg{}
+	for _, pack := range r.SrcPackages {
+		for _, binpkg := range pack.BinaryNames {
+			binToSrc[binpkg] = srcpkg{
+				name:    pack.Name,
+				version: pack.Version,
+				arch:    pack.Arch,
+			}
+		}
+	}
+
+	packages := make([]spdx.Package, 0, len(r.Packages))
+	for _, pack := range r.Packages {
+		var annotations []spdx.Annotation
+		if sp, ok := binToSrc[pack.Name]; ok {
+			annotations = appendAnnotation(annotations, "SrcName", sp.name, r.ReportedAt)
+			annotations = appendAnnotation(annotations, "SrcVersion", sp.version, r.ReportedAt)
+			annotations = appendAnnotation(annotations, "SrcArch", sp.arch, r.ReportedAt)
+		}
+
+		var externalRefs []*spdx.PackageExternalReference
+		purl := osPkgToPURL(r.Family, r.Release, pack)
+		externalRefs = appendExternalRefs(externalRefs, categoryPackageManager, packageManagerPURL, purl.String())
+
+		if urls, exists := packageToURLMap[pack.Name]; exists {
+			for _, url := range urls {
+				externalRefs = appendExternalRefs(externalRefs, categorySecurity, securityAdvisory, url)
+			}
+		}
```

Comment:
If you are just looping through urls, you don't need to evaluate exists.
```suggestion
		for _, url := range packageToURLMap[pack.Name] {
			externalRefs = appendExternalRefs(externalRefs, categorySecurity, securityAdvisory, url)
		}
```